### PR TITLE
Trash and Recent categories

### PR DESCRIPTION
**Pull request by somebody1234**
### Pull Request Description
- Closes https://github.com/enso-org/cloud-v2/issues/617
  - Trash category
  - Do not open "confirm delete" modal when on remote (cloud) backend
    - Permanent deletion has been superseded by the trash category
  - Save current category to `localStorage`
- Also implements "Recent" category. Differences to Home category:
  - Hides sort indicators
  - Disables sorting

Other fixes:
- Allow overriding cloud environment through environment variable
- Prevent triggering rename on click when multiple items are selected
- Remove "move to trash" context menu option when user is not owner

### Important Notes
- This registers the "Ctrl + R" (Cmd + R) shortcut for restoring an item from trash. This is the same shortcut used for renaming, however I think it is fine because items in trash cannot be renamed.
(As usual though, this is very simple to remove/change so it's not a big deal to remove/change it.)

- The "Recent" category lacks backend support - so it falls back to the default category, as the deserialization of unknown values falls back to the default category.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] ~~The documentation has been updated, if necessary.~~
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] ~~Unit tests have been written where possible.~~
  - [x] ~~If GUI codebase was changed, the GUI was tested when built using `./run ide build`.~~

### CODE REVIEW

**Comment by PabloBuchu** [comment]
removing a project still uses old api endpoint

File: app/ide-desktop/.vscode/react.code-snippets [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+            "}"
+        ]
+    },
+    "useState": {
+        "prefix": ["$s", "usestate"],
+        "body": [
+            "const [$1, set${1/(.*)/${1:/pascalcase}/}] = React.useState($2)"
+        ]
+    }
+}
```
Comment (PabloBuchu) [inline app/ide-desktop/.vscode/react.code-snippets]: no nl

**Comment by sylwiabr** [comment]
@PabloBuchu @indiv0 anything blocking the review here?

**Comment by somebody1234** [comment]
@sylwiabr I don't think so - but the backend functionality this relies on, has not yet been merged

**Comment by PabloBuchu** [comment]
@somebody1234 small change. to get recent projects we issue `GET /directories?recent_projects=true` without `parent_id` (it will return 400 if both set)

**Comment by somebody1234** [comment]
@PabloBuchu implemented, but not sure it's working correctly:
- recent category shows as empty (at least this means that it's aware of the `recent_projects` param)
- trash category shows the same items as home category

**Comment by PabloBuchu** [comment]
last one small fix I guess. The context menu is not being closed after you `Move to trash` also delete disappeared from optinos from when tring to remove local projects.. is this some kind of regression?

**Comment by somebody1234** [comment]
i think it may be a regression, yeah. luckily (?) i don't think the `Move to trash` not closing the context menu happens on `develop`, will investigate and fix soon

**Comment by somebody1234** [comment]
seems like the "Delete" option is also present on develop. not sure what happened here, maybe a bad merge

**Comment by PabloBuchu** [comment]
@somebody1234 please fix conflicts and lets merge it.

**Comment by PabloBuchu** [comment]
just a minor.. can we disable `drafts` and `root`? for now we wont be using them

**Review by PabloBuchu — APPROVED** [review]

**Pull request merged**